### PR TITLE
Invariant Improvements: Position rewards logging for multiple pools

### PR DESCRIPTION
**Pull request by prat-gpt**
# Description of change
## High level
* Update position and rewards manager for multiple pools.
* Fix regression test `test_regression_fenwick_index_2` to run with any token precision and quote token limits.

### CODE REVIEW

File: tests/forge/invariants/PositionsAndRewards/handlers/PositionPoolHandler.sol [hunk 1]
```diff
@@ -227,24 +227,29 @@ abstract contract PositionPoolHandler is UnboundedPositionPoolHandler {
...
-
-            // loop over tokenIds in bucket indexes
-            uint256[] memory tokenIds = getTokenIdsByBucketIndex(address(_pool), bucketIndex);
-            for (uint256 k = 0; k < tokenIds.length; k++) {
-                uint256 tokenId = tokenIds[k];
-                
-                uint256 posLp = _positionManager.getLP(tokenId, bucketIndex);
-                string memory tokenIdStr = string.concat("tokenID ", Strings.toString(tokenId));
-                printLog(string.concat(tokenIdStr, " LP in positionMan = "), posLp);
+        for (uint256 i = 0; i < _pools.length; i++) {
```
Comment (ith-harvey) [inline tests/forge/invariants/PositionsAndRewards/handlers/PositionPoolHandler.sol:231]: suggest adding a comment above -> `// loop over pools`

File: tests/forge/invariants/PositionsAndRewards/handlers/PositionPoolHandler.sol [hunk 2]
```diff
@@ -227,24 +227,29 @@ abstract contract PositionPoolHandler is UnboundedPositionPoolHandler {
...
+                for (uint256 k = 0; k < tokenIds.length; k++) {
+                    uint256 tokenId = tokenIds[k];
+                    
+                    uint256 posLp = _positionManager.getLP(tokenId, bucketIndex);
+                    string memory tokenIdStr = string.concat("tokenID ", Strings.toString(tokenId));
+                    printLog(string.concat(tokenIdStr, " LP in positionMan = "), posLp);
+                }
             }
         }
+
```
Comment (ith-harvey) [inline tests/forge/invariants/PositionsAndRewards/handlers/PositionPoolHandler.sol]: NIT: extra space

File: tests/forge/regression/ERC20Pool/RegressionTestBasicERC20Pool.t.sol [hunk 3]
```diff
@@ -3,6 +3,7 @@
 pragma solidity 0.8.18;
 
 import { BasicERC20PoolInvariants } from "../../invariants/ERC20Pool/BasicERC20PoolInvariants.t.sol";
+import "@std/console.sol";
```
Comment (ith-harvey) [inline tests/forge/regression/ERC20Pool/RegressionTestBasicERC20Pool.t.sol]: Left over console log

**Review by ith-harvey — APPROVED** [review]
Some cleanup, otherwise looks good

**Review by EdNoepel — APPROVED** [review]

File: tests/forge/invariants/PositionsAndRewards/handlers/PositionPoolHandler.sol (hunk 1 above)
Comment (prat-gpt) [reply tests/forge/invariants/PositionsAndRewards/handlers/PositionPoolHandler.sol:231]: Added in [208ca7b](https://github.com/ajna-finance/contracts/pull/931/commits/208ca7be525d696d7d051acd282bd67c67efb94b)

File: tests/forge/invariants/PositionsAndRewards/handlers/PositionPoolHandler.sol (hunk 2 above)
Comment (prat-gpt) [reply tests/forge/invariants/PositionsAndRewards/handlers/PositionPoolHandler.sol]: Fixed in [208ca7b](https://github.com/ajna-finance/contracts/pull/931/commits/208ca7be525d696d7d051acd282bd67c67efb94b)

File: tests/forge/regression/ERC20Pool/RegressionTestBasicERC20Pool.t.sol (hunk 3 above)
Comment (prat-gpt) [reply tests/forge/regression/ERC20Pool/RegressionTestBasicERC20Pool.t.sol]: Fixed in [208ca7b](https://github.com/ajna-finance/contracts/pull/931/commits/208ca7be525d696d7d051acd282bd67c67efb94b)

**Review by MikeHathaway — APPROVED** [review]
LGTM

**Pull request merged**